File: 250207/그 요일은/the-day-of-the-day.py
m1, d1, m2, d2 = map(int, input().split())
A = input()

num_of_days = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
day_of_week = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]

# 2024년 1월 1일 기준 계산
# 2024년 m1월 d1일은 월요일이라 가정
days1 = sum(num_of_days[:m1]) + d1
days2 = sum(num_of_days[:m2]) + d2
days = days2 - days1 + 1

cnt = days//7
if A in day_of_week[:days%7]:
    cnt += 1

print(cnt)

File: trail2/그 요일은/the-day-of-the-day.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();
        String A = sc.next();

        int[] day = new int[] {0,31,29,31,30,31,30,31,31,30,31,30,31};
        int day1 = 0;
        int day2 = 0;
        for (int i=1; i<=12; i++) {
            if (i < m1) {
                day1 += day[i];
            }
            if (i < m2) {
                day2 += day[i];
            }
        }
        day1 += d1;
        day2 += d2;

        String[] d = new String[] {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
        int diff = day2 - day1 + 1;
        int cnt = 0;
        for (int i=0; i<diff; i++) {
            if (d[i%7].equals(A)) {
                cnt++;
            }
        }
        // System.out.println(diff);
        System.out.println(cnt);
    }
}